src/rgw/rgw_rados.cc

```diff
@@ -10053,12 +10067,19 @@ int RGWRados::Object::Read::read(int64_t ofs, int64_t end, bufferlist& bl)
         pbl = &read_bl;
       }
     }
+  } else {
+    store->open_pool_ctx(read_obj.pool, tail_io_ctx);
+    tail_io_ctx.locator_set_key(read_obj.loc);
   }
 
   ldout(cct, 20) << "rados->read obj-ofs=" << ofs << " read_ofs=" << read_ofs << " read_len=" << read_len << dendl;
   op.read(read_ofs, read_len, pbl, NULL);
 
-  r = state.io_ctx.operate(read_obj.oid, &op, NULL);
+  if (reading_from_head)
```

Comment:
it is not related to manifest.

we have two pools now, and will also have two rados pool ioctx. so, which rados ioctx should we use when code reach here? we can do the choice according to whether it is head object or not.